schemas/config.json

```diff
@@ -0,0 +1,367 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/eth-cscs/reframe/master/schemas/config.json",
+    "title": "Validation schema for ReFrame's configuration file",
+    "defs": {
+        "alphanum_string": {
+            "type": "string",
+            "pattern": "([a-zA-Z0-9_]|-)+"
+        },
+        "system_ref": {
+            "type": "array",
+            "items": {"type": "string"}
+        },
+        "envvar_list": {
+            "type": "array",
+            "items": {
+                "type": "array",
+                "items": [
+                    {
+                        "type": "string",
+                        "pattern": "([a-zA-Z_][a-zA-Z0-9_]*)"
+                    },
+                    {"type": "string"}
+                ],
+                "additionalProperties": false
+            }
+        },
+        "modules_list": {
+            "type": "array",
+            "items": {"type": "string"}
+        },
+        "loglevel": {
+            "type": "string",
+            "enum": ["critical", "error", "warning",
+                     "info", "verbose", "debug"]
+        },
+        "handler_common": {
+            "type": "object",
+            "properties": {
+                "type": {"type": "string"},
+                "level": {"$ref": "#/defs/loglevel"},
+                "format": {"type": "string"},
+                "datefmt": {"type": "string"}
+            },
+            "required": ["type"]
+        },
+        "file_handler": {
+            "allOf": [
+                {"$ref": "#/defs/handler_common"},
+                {
+                    "properties": {
+                        "name": {"type": "string"},
+                        "append": {"type": "boolean"},
+                        "timestamp": {
+                            "anyOf": [
+                                {"type": "boolean"},
```

Comment:
I do not understand why the `timestamp` can be a boolean.

Reply:
This is exactly what we specify in documentation. It's similar to a command-line option with an optional argument. If `true`, you will use the default format, if you pass the format, you will use that one.